### ISSUE TITLE
Some emojis are clipped, when rendered with a gradient fill

### ISSUE THREAD

**Issue by lukasrt** [typst/typst#3755]
### Description

When using a gradient as text fill, large emojis do not display correctly as parts are being clipped.

Minimal example:

```
// OK
#emoji.house

// Not OK, clipped
#text(fill: gradient.linear(..color.map.crest), emoji.house)
```

Result:
![grafik](https://github.com/typst/typst/assets/12510975/63875968-d0fe-404c-a1aa-2013550ba452)

### Operating system

Windows

### Typst version

- [X] I am using the latest version of Typst

**Comment by Enivex** [comment]
@elegaanz would this have been affected by all the recent emoji work, or is that purely for PDF?

**Comment by elegaanz** [comment]
Normally it only affected PDF export, yes. There have been a bit of refactoring of emoji handling in general, so it may have changed the behavior of other export formats as well, but this wasn't intended if it happened.

I wanted to check if I could reproduce this bug, but I can't because I don't know what font was originally used.

**Comment by Enivex** [comment]
> Hello, the font in use is Linux Libertine, though the same behaviour can be seen for all fonts with emojies, that exceed a certain height, if I recall correctly. The screenshot is taken from a PDF export. I checked with version 0.11.1 and the issue still persists.

Linus Libertine is not an emoji font, so it's likely fallback.

> The screenshot is taken from a PDF export. I checked with version 0.11.1 and the issue still persists.

PDF won't have emoji support until 0.12

**Comment by 411A** [comment]
Hi, I am experiencing the same problem. The Segoe UI Emoji font works fine, but the Noto Color Emoji font clips the right side of the emojis. I tried different font sizes and inserting the text inside a box, but it didn't fix the issue.
![image](https://github.com/typst/typst/assets/13054619/74626e3b-bc87-48dd-9112-61283b9f5b79)

**Comment by Enivex** [comment]
> Hi, I am experiencing the same problem. The `Segoe UI Emoji` font works fine, but the `Noto Color Emoji` font clips the right side of the emojis. I tried different font sizes and inserting the text inside a box, but it didn't fix the issue.
> Code:
> ```
> #text(font: "Segoe UI Emoji", size: 20pt, fallback: false, dir: ltr)[#emoji.chain]
> #text(font: "Noto Color Emoji", size: 20pt, fallback: false, dir: ltr)[#emoji.chain]
> ```
> ![Image of exported PDF from Typst](https://github.com/typst/typst/assets/13054619/74626e3b-bc87-48dd-9112-61283b9f5b79)
> 

Is this on main?

**Comment by 411A** [comment]
> Is this on main?

Yes, it's on `main.typ`. The image is taken from the preview generated by the VSCode extension [Typst Preview](https://marketplace.visualstudio.com/items?itemName=mgt19937.typst-preview), compiled by the command `typst compile main.typ` on version `typst 0.11.1 (50115102)`.

**Comment by terefang** [comment]
looking on the metrics of the two fonts i whould expect that the SVGs viewport could cause the clipping.

![Image](https://github.com/user-attachments/assets/bcaf9927-0410-4c7e-9d3d-b5ab5509b1ec)
